### ISSUE TITLE
ERROR when dealing with single-label domain

### ISSUE THREAD

**Issue by CUHKJason** [gentilkiwi/mimikatz#240]
Hi, recently I am using mimikatz on a single-label domain (i.e. \testing).
First I tried to perform dcsync using lsadump::dcsync /domain:testing ...
It failed with error message:

> Domain not present, or doesn't look like a FQDN

I found the reason is in line 47 of kuhl_m_lsadump_dc.c
 `if(szDomain && wcschr(szDomain, L'.'))`
It was success after I changed it into `if(szDomain)`

However, I encountered the similar error while performing golden ticket attack:

> Domain name does not look like a FQDN

It's from the if-then statement in line 430 of kuhl_m_kerberos.c
`if(baseDot = wcschr(szDomain, L'.'))`
as the variable baseDot is needed from line 464, may I know if there's any quick fix so that I could generate the ticket while specifying /domain:testing?

**Comment by edermi** [comment]
Did you try appending a dot to your domain, e.g. `testing.`? Technically, this is a valid FQDN.

**Comment by CUHKJason** [comment]
Yes it's a valid FQDN and I'm able to generate the ticket but the domain would be `testing.` instead of `testing` so it doesn't work...
